app/models/trainee.rb

```diff
@@ -202,6 +202,10 @@ class Trainee < ApplicationRecord
   before_save :clear_lead_school_id, if: :lead_school_not_applicable?
   before_save :set_submission_ready, if: :completion_trackable?
 
+  def academic_cycle
```

Comment:
I don't think we need this or at least we don't have a use case for it at the moment. If we do 'in the future :tm:' I wonder if, as `AcademicCycle` already has some knowledge of `Trainee` and of the trainee start | course start logic it would be better implemented as something like `AcademicCycle.start_cycle_for(trainee: t)` Having it here means that precedence between start dates is defined in two places. I reckon we can just get rid of this for now though and cross that bridge when we come to it. If we do get rid we could mention this in the ADR.